leshan-server-core/src/main/java/org/eclipse/leshan/server/queue/impl/QueuedRequestSender.java

```diff
@@ -0,0 +1,282 @@
+        if (isQueueMode(destination.getBindingMode())) {
+            LOG.debug("Sending request {} with queue mode", castedDownlinkRequest);
+
+            final String endpoint = destination.getEndpoint();
+
+            // Accept messages only when the client is already known to ClientRegistry
+            if (clientRegistry.get(endpoint) != null) {
+                final QueuedRequest queuedRequest = queuedRequestFactory.newQueueRequestEntity(endpoint,
+                        castedDownlinkRequest);
+                // prepare error and response context
+                responseContextHolder.put(queuedRequest.getRequestId(), new ResponseContext() {
+                    @Override
+                    public ResponseCallback<LwM2mResponse> getResponseCallback() {
+                        // we rely that we get called with correct response
+                        @SuppressWarnings("unchecked")
+                        ResponseCallback<LwM2mResponse> castedResponseCallback = (ResponseCallback<LwM2mResponse>) responseCallback;
+                        return castedResponseCallback;
+                    }
+
+                    @Override
+                    public ErrorCallback getErrorCallback() {
+                        return errorCallback;
+                    }
+                });
+
+                messageStore.add(queuedRequest);
+                // If Client is reachable and this is the first message, we send it immediately.
+                if (clientStatusTracker.startClientReceiving(endpoint)) {
+                    processingExecutor.execute(newRequestSendingTask(endpoint));
+                }
+            } else {
+                LOG.warn("Ignoring a message received in Queue Mode for the unknown client [{}]", endpoint);
+            }
+        } else {
+            // send directly (client is not in queued mode)
+            delegateSender.send(destination, request, responseCallback, errorCallback);
+        }
+    }
+
+    @Override
+    public void stop() {
+        clientRegistry.removeListener(queueModeClientRegistryListener);
+        observationRegistry.removeListener(queueModeObservationRegistryListener);
+        processingExecutor.shutdown();
+        try {
+            boolean queueProcessingExecutorTerminated = processingExecutor.awaitTermination(5, TimeUnit.SECONDS);
+            if (!(queueProcessingExecutorTerminated)) {
+                LOG.debug(
+                        "Could not stop all executors within timeout. processingExecutor stopped: {}, ",
+                        queueProcessingExecutorTerminated);
+            }
+        } catch (InterruptedException e) {
+            LOG.debug("Interrupted while stopping. Abort stopping.");
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    private RequestSendingTask newRequestSendingTask(final String endpoint) {
+        return new RequestSendingTask(clientRegistry, delegateSender, processingExecutor,
```

Comment:
Thought of having the try/catch around run() instead of each executable Task. Anyways, we have only two tasks in Queue mode and I can remove, when this ExceptionAwareExecutorService is not used anywhere else in leshan.